File: TEST_javascript/fibonacci.js
// Time complexity = O(n), Space complexity = O(1). Multi-variable loop with simple addition, IGNORE THIS CODE
function Fibonacci(n) {
    let a = BigInt(0), b = BigInt(1), temp = BigInt(0);
    for (let i = 0; i < n; i++) {
        temp = a;
        a = b;
        b = temp + b;
    }
    return a;
}
    
Fibonacci(1000000);

File: TEST_python/fibonacci.py
# Function for nth Fibonacci number, Time complexity = O(n), Space complexity = O(1). Multi-variable loop with simple addition
def Fibonacci(n):

	a, b, c = 0, 1, 0
	for i in range(0, n):
		c = a
		a = b
		b = c + b
	return a

# Driver Program
Fibonacci(1000000) #1000000 = default